mongodbatlas/federation_settings_connected_org_config.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const federationSettingsConnectedOrganizationBasePath = "api/atlas/v1.0/federationSettings/%s/connectedOrgConfigs"
+
+// A Resource describes a specific resource the Role will allow operating on.
+
+// FederatedSettings represents a FederatedSettings Connected Organization.
+type FederatedSettingsConnectedOrganizations struct {
+	Links      []*Link                                   `json:"links,omitempty"`
+	Results    []*FederatedSettingsConnectedOrganization `json:"results,omitempty"`
+	TotalCount int                                       `json:"totalCount,omitempty"`
+}
+
+type FederatedSettingsConnectedOrganization struct {
+	DomainAllowList          []string        `json:"domainAllowList,omitempty"`
+	DomainRestrictionEnabled *bool           `json:"domainRestrictionEnabled,omitempty"`
+	IdentityProviderID       string          `json:"identityProviderId,omitempty"`
+	OrgID                    string          `json:"orgId,omitempty"`
+	PostAuthRoleGrants       []string        `json:"postAuthRoleGrants,omitempty"`
+	RoleMappings             []*RoleMappings `json:"roleMappings,omitempty"`
+	UserConflicts            *UserConflicts  `json:"userConflicts,omitempty"`
+}
+
+type RoleMappings struct {
+	ExternalGroupName string             `json:"externalGroupName,omitempty"`
+	ID                string             `json:"id,omitempty"`
+	RoleAssignments   []*RoleAssignments `json:"roleAssignments,omitempty"`
+}
+
+// List gets all Federated Settings Connected Organization (Org-Mappings).
+//
+// See more: https://www.mongodb.com/docs/atlas/reference/api/org-mappings-return-all/
+func (s *FederatedSettingsServiceOp) ListConnectedOrgs(ctx context.Context, opts *ListOptions, federationSettingsID string) (*FederatedSettingsConnectedOrganizations, *Response, error) {
```

Comment:
let's please keep list options always last 
```suggestion
func (s *FederatedSettingsServiceOp) ListConnectedOrgs(ctx context.Context, federationSettingsID string, opts *ListOptions,) (*FederatedSettingsConnectedOrganizations, *Response, error) {
```

Reply:
Refactored